ctx_test.go

```diff
@@ -164,11 +174,21 @@ func TestContext_QueryParam(t *testing.T) {
 
 		paramBool, err = c.QueryParamBoolErr("notfound")
 		require.Error(t, err)
-		require.False(t, paramBool)
+		assert.Zero(t, paramBool)
+		notFoundErr := &internal.QueryParamNotFoundError{}
+		require.ErrorAs(t, err, notFoundErr)
+		assert.Equal(t, "param notfound not found", notFoundErr.Error())
+		assert.Equal(t, http.StatusUnprocessableEntity, notFoundErr.StatusCode())
+		assert.Equal(t, "param notfound not found", notFoundErr.DetailMsg())
 
 		paramBool, err = c.QueryParamBoolErr("other")
 		require.Error(t, err)
-		require.False(t, paramBool)
+		require.Zero(t, paramBool)
+		invalidErr := &internal.QueryParamInvalidTypeError{}
+		require.ErrorAs(t, err, invalidErr)
```

Comment:
I would like to suggest writting this


```suggestion
		var invalidErr internal.QueryParamInvalidTypeError
		require.ErrorAs(t, err, &invalidErr)
```

The pointer is only needed for errors.As

Reply:
Ah yeah this broke the tests with `panic: errors: target must be a non-nil pointer`.

Reply:
Also testifylint

Reply:
and then if like this

```
var invalidErr internal.QueryParamInvalidTypeError
require.ErrorAs(t, err, &invalidErr)
```

We get:

```
                Error Trace:    /Users/dhitt/github/go-fuego/fuego/ctx_test.go:145
                Error:          Should be in error chain:
                                expected: "query param = is not of type : %!s(<nil>)"
                                in chain: "param notfound not found"
                Test:           TestContext_QueryParam/int
```

Reply:
ope nevermind im dumb

Reply:
Take a co author for that 😓

Reply:
So it cannot work here with var errWhatever WhateverError + ErrorAs(t, err, &errWhatever) ?

Reply:
Na it does that is what I ended up with

```
https://github.com/go-fuego/fuego/pull/494/files#diff-593b58937cba58a2e213b636a0c60bd7671faba0fd51d12a59851ca6d7fec6a3R153
```

I missed the `&`